extends Control
signal request_switch_to_TOP_CONTROL(page_name: String)
func _ready():
	print("âœ… Main_control å•Ÿå‹•ï¼ˆCamera æ¨¡å¼ï¼‰")

	$NoticeItem_control.request_switch_to.connect(_on_request_switch_to)
	$Task_control.request_switch_to.connect(_on_request_switch_to)
	$Jokes_control.request_switch_to.connect(_on_request_switch_to)
	$StoreRoom_contorl.request_switch_to.connect(_on_request_switch_to)

	_show_camera("StoreRoom_Camera2D")
	#_show_camera("Task_Camera2D")

func _on_request_switch_to(target: String) -> void:
	match target:
		"Task_control":
			print("ðŸ” åˆ‡æ›åˆ°ä»»å‹™é é¢ CameraTask")
			_show_camera("Task_Camera2D")
		"Jokes_control":
			print("ðŸ” åˆ‡æ›åˆ°ç¬‘è©±é é¢ CameraJokes")
			_show_camera("Jokes_Camera2D")
		"StoreRoom_control":
			print("ðŸ” åˆ‡æ›åˆ°å€‰åº«é é¢ CameraStoreRoom")
			_show_camera("StoreRoom_Camera2D")
		"NoticeItem_control":
			print("ðŸ” å›žåˆ°é¦–é  CameraNotice")
			_show_camera("NoticeItem_Camera2D")
		"Home":
			emit_signal("request_switch_to_TOP_CONTROL","home")
			print("é€™è£¡è¦åˆ‡æ›åˆ°ä¸»é é¢")
			get_tree().quit()
		"Farm_control":
			emit_signal("request_switch_to_TOP_CONTROL","farm")
			print("ðŸ”é€™è£¡è¦åˆ‡æ›åˆ°Farmé é¢")
		_:
			print("âš ï¸ æœªçŸ¥æŒ‡ä»¤ï¼š", target)

func _show_camera(camera_name: String) -> void:
	for node in get_children():
		if node is Camera2D:
			node.enabled = (node.name == camera_name)
